File: Atcoder/abc057/b.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,N) for(int i=0;i<int(N);++i)
#define rep1(i,N) for(int i=1;i<int(N);++i)
#define all(a) (a).begin(),(a).end()

typedef long long ll;
typedef vector<int> vi;
typedef set<int> seti;
typedef vector<string> vs;

const int MOD = 1e9 + 7;
const int inf = 1e9;

int main() {
	ll N,M;
	cin>>N>>M;
	vector<ll> a(N),b(N),c(M),d(M),D(N),ans(N);
	rep(i,N)cin>>a[i]>>b[i];
	rep(i,M)cin>>c[i]>>d[i];
	rep(i,N){
		D[i]=inf;
		rep(j,M){
			ll dis = abs(a[i]-c[j])+abs(b[i]-d[j]);
			if(D[i]>dis){
				ans[i]=j+1;
				D[i] = dis;
			}
		}
	}
	rep(i,N)
	cout<<ans[i]<<endl;
}

File: atcoder/abc057/b.py
N,M = map(int,input().split())
A = []
B = []
for i in range(N):
	a,b = map(int,input().split())
	A.append([a,b])
for i in range(M):
	c,d = map(int, input().split())
	B.append([c,d])
#print(A,B)

for i in range(N):
	z = A[i]
	ans = []
	for j in range(M):
		ch = B[j]
		d = abs(z[0]-ch[0]) + abs(z[1]-ch[1])
		ans.append([j,d])
	#print(ans)
	ans.sort(key=lambda x:x[1])
	#print(ans)
	print(ans[0][0]+1)

